lib/src/util.dart

```diff
@@ -611,3 +612,13 @@ Iterable<String> allDartPathsExceptHidden() =>
 Iterable<String> allDartPathsExceptHiddenAndGenerated() =>
     filePathsFromGlob(Glob('**.dart', recursive: true))
         .where((path) => !path.endsWith('.g.dart'));
+
+/// Returns whether or not [sourceText] contains invalid code and, therefore,
+/// should be skipped when generating patches.
+///
+/// Used to override [Suggestor.shouldSkip] when parsing errors should be
+/// skipped instead of thrown when [Suggestor.generatePatches] is called.
+bool shouldSkipParsingErrors(String sourceText) {
```

Comment:
This name might imply that it's making a determination, based on the contents of a file, whether it's okay to skip parsing errors.

I think a better name for this function would be
```suggestion
bool hasParseErrors(String sourceText) {
```
which more clearly states what this function is doing, and also makes it clearer at the call-sites why exactly we're skipping a file.

```dart
bool shouldSkip(String sourceText) => hasParseErrors(sourceText);
```